Datastore/tests/System/RunTransactionTest.php

```diff
@@ -79,6 +81,61 @@ public function testRunTransactions(DatastoreClient $client)
         $this->assertEquals($newLastName, $result['lastName']);
     }
 
+    /**
+     * @dataProvider multiDbClientProvider
+     */
+    public function testRunAggregationQueryWithTransactions(DatastoreClient $client)
+    {
+        $this->skipEmulatorTests();
+        $kind = 'Person';
+        $testId = rand(1, 999999);
+        $key1 = $client->key($kind, $testId);
+        $data = ['lastName' => 'Smith'];
+        $newLastName = 'NotSmith';
+        $entity1 = $client->entity($key1, $data);
+
+        $transaction = $client->transaction();
+        $transaction->insert($entity1);
+        $transaction->commit();
+
+        self::$localDeletionQueue->add($key1);
+
+        // validate default DB should not have data
+        $defaultDbClient = current(self::defaultDbClientProvider())[0];
+        $this->assertOtherDbEntities($defaultDbClient, $kind, $testId, 0);
+
+        // transaction with query
+        $transaction2 = $client->transaction();
+        $query = $client->query()
+            ->kind($kind)
+            ->hasAncestor($key1)
+            ->filter('lastName', '=', 'Smith');
+        $results = iterator_to_array($transaction2->runQuery($query));
+
+        $this->assertAggregationQueryResult($transaction2, $query, 1);
+
+        $results[0]['lastName'] = $newLastName;
+        $transaction2->update($results[0]);
+        $transaction2->commit();
+
+        $this->assertCount(1, $results);
+
+        // read transaction with aggregation query
+        $transaction3 = $client->readOnlyTransaction();
+        $query = $client->query()
+        ->kind($kind)
+        ->hasAncestor($key1)
+        ->filter('lastName', '=', 'Smith');
+        $this->assertAggregationQueryResult($transaction3, $query, 0);
+
+        // transaction with lookup
+        $transaction4 = $client->transaction();
+        $result = $transaction4->lookup($key1);
+        $transaction4->rollback();
```

Comment:
Is this test code realted to aggregations?🤔

Reply:
Removed, nice catch :-)